s3/src/main/scala/akka/stream/alpakka/s3/auth/Signer.scala

```diff
@@ -0,0 +1,55 @@
+package akka.stream.alpakka.s3.auth
+
+import java.security.MessageDigest
+import java.time.format.DateTimeFormatter
+import java.time.{ ZoneOffset, ZonedDateTime }
+
+import akka.http.scaladsl.model.headers.RawHeader
+import akka.http.scaladsl.model.{ HttpHeader, HttpRequest }
+import akka.stream.Materializer
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+object Signer {
+  private val dateFormatter = DateTimeFormatter.ofPattern("YYYYMMdd'T'HHmmssX")
+
+  def signedRequest(request: HttpRequest, key: SigningKey, date: ZonedDateTime = ZonedDateTime.now(ZoneOffset.UTC))(implicit mat: Materializer): Future[HttpRequest] = {
+    implicit val ec: ExecutionContext = mat.executionContext
+    val hashedBody = request.entity.dataBytes.runWith(StreamUtils.digest()).map {
+      case hash =>
+        Utils.encodeHex(hash.toArray)
+    }
+
+    hashedBody.map {
+      case hb =>
+        val headersToAdd = Seq(RawHeader("x-amz-date", date.format(dateFormatter)), RawHeader("x-amz-content-sha256", hb)) ++ sessionHeader(key.credentials)
```

Comment:
Please be specific when creating a `Seq` instance, i.e. use `List` or `Vector` (there are tradeoffs depending on the use case).